config/file_test.go

```diff
@@ -436,6 +436,31 @@ func TestFileStoreLoad(t *testing.T) {
 		assert.Equal(t, map[string]interface{}{"ServiceSettings": map[string]interface{}{"SiteURL": true}}, fs.GetEnvironmentOverrides())
 	})
 
+	t.Run("do not persist environment variables", func(t *testing.T) {
+		path, tearDown := setupConfigFile(t, minimalConfig)
+		defer tearDown()
+
+		fs, err := config.NewFileStore(path, false)
+		require.NoError(t, err)
+		defer fs.Close()
+
+		assert.Equal(t, "http://minimal", *fs.Get().ServiceSettings.SiteURL)
+
+		os.Setenv("MM_SERVICESETTINGS_SITEURL", "http://override")
```

Comment:
I was wrong to Set first: there's no point, since the env variable doesn't have a chance to override until Load. Should have called it after. 

But, I'm pretty sure line 447 checks that the environment override isn't being persisted from the prior test. Either way, I'll make it unambiguous by make the env variable unique for this test -- good idea. And I'll move it to before the file store. (I'll do that for the previous test as well.)  Now there's no reason call load a second time at all.

Reply:
I think it's fine to narrow this test to focus on the not writing to disk part.